test/cpp/end2end/server_interceptors_end2end_test.cc

```diff
@@ -44,7 +44,32 @@ namespace {
 
 class LoggingInterceptor : public experimental::Interceptor {
  public:
-  LoggingInterceptor(experimental::ServerRpcInfo* info) { info_ = info; }
+  LoggingInterceptor(experimental::ServerRpcInfo* info) {
+    info_ = info;
+
+    // Check the method name and compare to the type
+    const char* method = info->method();
+    experimental::ServerRpcInfo::Type type = info->type();
+
+    // Check that we use one of our standard methods with expected type.
+    // We accept BIDI_STREAMING for Echo in case it's an AsyncGenericService
```

Comment:
Would it make sense to do this conditionally based on which test is being run?  Or is it too hard to instrument that?

Reply:
IMO, instrumenting that would break the clean separation between the client-side which initiates the test and the server side which is being intercepted here. So, I'd prefer not to do so.